ratis-test/src/test/java/org/apache/ratis/datastream/TestDataStreamDisabled.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ratis.datastream;
+
+import org.apache.ratis.RaftConfigKeys;
+import org.apache.ratis.conf.RaftProperties;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class TestDataStreamDisabled extends TestDataStreamBase {
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+  @Before
+  public void setup() {
+    properties = new RaftProperties();
+    RaftConfigKeys.DataStream.setType(properties, SupportedDataStreamType.DISABLED);
+  }
+
+  @Test
+  public void testDataStreamSingleServer() throws Exception {
+    exception.expect(UnsupportedOperationException.class);
+    runTestDataStream(1, 1_000_000, 100);
+    runTestDataStream(1,1_000, 10_000);
+  }
```

Comment:
It is not clear when we get UnsupportedOperationException.  In the test, we should make sure that server and client can start and work normally.  When the client uses the StreamApi, it will get UnsupportedOperationException.

Reply:
@szetszwo 

I have push a one line of change to match the exception error message, which matches with the unsupported operation exception thrown from https://github.com/apache/incubator-ratis/blob/master/ratis-client/src/main/java/org/apache/ratis/client/DataStreamClientRpc.java#L34.

The error message might have proved that server and client have started working and it fails when call `streamAsync`. Do you think whether this is enough?

If this is not enough, another approach is to expose `setupServer` and `setupClient` and test cases call these one by one to identify which step stops.

Reply:
Actually have changed this test to the version that `setupServer` and `setupClient` then make a call.